File: src/main/java/com/dh/ondot/schedule/infra/dto/NaverSearchPlaceResponse.java
package com.dh.ondot.schedule.infra.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public record NaverSearchPlaceResponse(
        @JsonProperty("items") List<NaverPlace> places
) {
    public record NaverPlace(
            String title,
            String roadAddress,
            Integer mapx,
            Integer mapy
    ) {}
}

File: src/main/kotlin/com/dh/ondot/schedule/infra/dto/NaverSearchPlaceResponse.kt
package com.dh.ondot.schedule.infra.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class NaverSearchPlaceResponse(
    @param:JsonProperty("items") val places: List<NaverPlace>,
) {
    data class NaverPlace(
        val title: String,
        val roadAddress: String?,
        val mapx: Int,
        val mapy: Int,
    )
}
